src/main/groovy/com/github/jengelman/gradle/plugins/shadow/internal/UnusedTracker.groovy

```diff
@@ -58,10 +74,11 @@ class UnusedTracker {
         apiDependencies.each { dep ->
             if (dep instanceof ProjectDependency) {
                 apiJars.addAll(getApiJarsFromProject(dep.dependencyProject))
-                apiJars.add(runtimeConfiguration.find { it.name.endsWith("${dep.name}.jar") } as File)
+                addJar(runtimeConfiguration, dep, apiJars)
             } else if (dep instanceof SelfResolvingDependency) {
                 apiJars.addAll(dep.resolve())
             } else {
+                addJar(runtimeConfiguration, dep, apiJars)
```

Comment:
Why this and the next line? Ain't some duplication going on then?

Reply:
I assumed that the intent is essentially the same in both location (i.e., add only the dependency jar). Although, arguably the `isProjectDependencyFile` name makes less sense this way. If you mean the duplicate call to `addJar`, then it is difficult to not repeat something with the middle condition in the way.